python/lsst/analysis/drp/histPlot.py

```diff
@@ -274,94 +236,95 @@ def histPlot(self, catPlot, plotInfo, sumStats):
+        col_bounds = np.linspace(0+summary_block+panel_pad, 240, ncols+1, dtype=int)
+        row_bounds = np.linspace(0, 240, nrows+1, dtype=int)
+        col_starts, col_stops, row_starts, row_stops = [], [], [], []
+        for i in range(len(panels)):
+            col_starts.append(col_bounds[i%ncols]+panel_pad)
+            if (i == (len(panels)-1)) and (len(panels)%2 == 1):
+                col_stops.append(np.max(col_bounds)-panel_pad)
+            else:
+                col_stops.append(col_bounds[(i%ncols)+1]-panel_pad)
+            row_starts.append(row_bounds[i//ncols]+panel_pad)
+            row_stops.append(row_bounds[(i//ncols)+1]-(2*panel_pad))
+
+        # panel plotting
+        for i, (panel, col_start, col_stop, row_start, row_stop) in enumerate(zip(panels,
+                                                                                  col_starts, col_stops,
+                                                                                  row_starts, row_stops)):
+            hists = [x for x in panels[panel].toDict()['actions'].keys()]
+            columns = []
+            for hist in hists:
+                columns.append(f"p{i}_{hist}")
+
+            # panel plotting limits
+            vLowers, vUppers = [], []
+            for column in columns:
+                pvals = np.nanpercentile(datPlot[column].values,
+                                         [self.config.panels[panel].pLower, self.config.panels[panel].pUpper])
+                vLowers.append(pvals[0])
+                vUppers.append(pvals[1])
+            vLower = np.min(vLowers)
+            vUpper = np.max(vUppers)
+
+            # generate plot
+            ax = fig.add_subplot(gs[row_start:row_stop, col_start:col_stop])
+            for count, (hist, column) in enumerate(zip(hists, columns)):
+                isfinite = np.isfinite(datPlot[column])
+                datMed = np.nanmedian(datPlot[column][isfinite])
+                datMad = sigmaMad(datPlot[column][isfinite])
+                col = plt.cm.tab10(count)
+                histLabel = f"{hist}:::{datMed:0.1f}, {datMad:0.1f}, {np.sum(isfinite)}"
+                ax.hist(datPlot[column][isfinite], bins=self.config.panels[panel].nBins, alpha=0.7, color=col,
+                        range=(vLower, vUpper), histtype="bar", lw=2, label=histLabel)
+                ax.axvline(datMed, ls=':', lw=2, c=col)
+            ax.set_yscale(self.config.panels[panel].yscale)
+            ax.set_xlim(vLower, vUpper)
+            ylims = ax.get_ylim()
+            ax.set_ylim(ylims[0], ylims[1]*1.1)
+            ax.set_xlabel(self.config.panels[panel].label, labelpad=1)
+            ax.tick_params(labelsize=7)
+            # add legends
+            leg1 = ax.legend(loc="upper left", fontsize=7, frameon=False, handlelength=0, borderaxespad=0,
+                             handletextpad=0, handleheight=0.7, framealpha=0.4)
+            _ = plt.gca().add_artist(leg1)
```

Comment:
Do you need a return thing here?

Reply:
The entire legend section has now been refactored to something much simpler, using `ax.text`.